tests/phpunit/CRM/Utils/UrlTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Class CRM_Utils_UrlTest
+ * @group headless
+ */
+class CRM_Utils_URLTest extends CiviUnitTestCase {
+
+  public function setUp() {
+    parent::setUp();
+  }
+
+  /**
+   * Test the redirect hook.
+   *
+   * @param string $url
+   * @param array $parsedUrl
+   *
+   * @dataProvider getURLs
+   */
+  public function testRedirectHook($url, $parsedUrl) {
```

Comment:
I'm not sure if it makes sense for this test to be here. The `CRM_Utils_Url` class doesn't know anything about hooks or redirects. It only parses and unparses URLs. That's why I've mentioned [here](https://github.com/civicrm/civicrm-core/pull/11519/files#r167558562) this I think we should use this test class to only test the `CRM_Utils_Url` methods and do the redirect/hook tests in the `CRM_Utils_SystemTest` class

Reply:
hmm - well it DOES test the parse / unparse in that test. I can move it back but I think the test itself gives adequate cover of those

Reply:
OK - I've moved it back into the System test & ditched the UrlTest - I think the parsing / unparsing is covered enough by the test but it does make more sense to be in the headline tested class